src/UtilityFunctions.cpp

```diff
@@ -41,10 +41,19 @@ int sumGeneratedNumbers(std::unique_ptr<NumberGenerator> gen, int num)
 
 int sumNumbersFromGenerators(std::vector<GeneratorType> const & types, int num)
 {
-    // TODO: Write a loop that goes over a collection of GeneratorType values (types)
-    // and creates generator for given type (use createGenerator function),
-    // then sums up all generated values for each type and returns the sum of sums.
-    // Use sumGeneratedNumbers to sum generate values for each generator.
-    return 42;
+    int retVal = 0;
+    try
+    {
+        for (auto type : types)
```

Comment:
It's recommened to use `const auto & type` to not create an additional copy here.